files/__addonLocation__/babel.config.json

```diff
@@ -1,5 +1,6 @@
 {
-  "plugins": [
+<% if (typescript) { %>  "presets": [["@babel/preset-typescript"]],
+<% } %>  "plugins": [
     "@embroider/addon-dev/template-colocation-plugin",
```

Comment:
before I forget, I think we'll want to change the default settings:

```
    [
      resolve('@babel/plugin-transform-typescript'),
      {
        allowDeclareFields: true,
        onlyRemoveTypeImports: true,
        // Default enums are IIFEs
        optimizeConstEnums: true,
      },
    ],
```

This is what I use over here: https://github.com/NullVoxPopuli/ember-popperjs/blob/main/ember-popperjs/babel.config.cjs#L8

and cc @chriskrycho

Reply:
> Why plugin-transform-typescript instead of preset-typescript as Simon currently has?

I think we need both -- preset-typescript for all TS things (and a general indicator to users, that we're using "TS stuff" for compilation -- and then the transform plugin to override declare fields.

> Why resolve()?

catches typos with better errors at build time -- this is a personal preference, and it's a non issue if you just don't misspell things -- though maybe the error reporting has gotten better.

> Why default users to onlyRemoveTypeImports and optimizeConstEnums? 

for optimizeConstEnums, I don't understand why anyone would want the previous behavior of an IIFE per-enum.

For `onlyRemoveTypeImports`, yeah, I think this is an artifact of a darker age in my babel config's history -- we don't need this one

Reply:
> I think we need both

The preset sets up the transform for you, and accepts the same configuration

> catches typos with better errors at build time

We aren't using `resolve` anywhere else in that file (and in fact we can't—it's `.json`). That feels like an unrelated change to this PR if you want to propose it.

> for optimizeConstEnums, I don't understand why anyone would want the previous behavior of an IIFE per-enum

Sure, but `const enum` is somewhat of a power user feature with a lot of documented caveats and downsides (not to mention [the "just use unions" crowd](https://www.bam.tech/article/should-you-use-enums-or-union-types-in-typescript)), so my argument here is that diverging from default behavior and introducing extra boilerplate config for maintainers isn't worth the tradeoff.

Someone who knows they have a highly performance-sensitive situation and that it means they want to use `const enum` is also likely to be able to determine that they want to change Babel's transpilation behavior as well. (Also, if you're in such a situation, actual `consts` + a union type will be more performant than `optimizeConstEnums`'s output, since it skips an object field traversal)